Gum/Controls/MainPanelControl.xaml.cs

```diff
@@ -34,9 +34,39 @@ IEnumerable<TabControl> AllControls
     public MainPanelControl()
     {
         InitializeComponent();
+
+        OverrideFontSizeProperty();
         this.KeyDown += HandleKeyDown;
     }
 
+    private void OverrideFontSizeProperty()
+    {
+        const double defaultFontSize = 11;
+
+        FontSizeProperty.OverrideMetadata(
```

Comment:
I do not understand what is happening here, this feels like magic.

Is this DI ?

This OverrideFontSizeProperty() is only called in the constructor, how is it handling font changes after the fact?

I assume that the "coerce" method is called whenever the font size changes for any of those 3 "reasons" you provided (AffectsMeasure,AffectsRender,Inherits?)

Reply:
FontSize is a dependency property of Control, I originally overwrote it to hook into a callback to trigger the changes in the tree view, but ended up handling that in the tree view itself... leaving just a coercion. This honestly could probably just be handled by the one invoking the change, and could probably be removed, but also it's not hurting anything.